src/components/core/ContextSelector/index.tsx

```diff
@@ -104,34 +104,52 @@ const ContextSelector: FC = () => {
 
             const onChangeQueryText = useCallback((e: ChangeEvent<HTMLInputElement>) => {
                 setQueryText(e.target.value);
+                setIsOpen(true);
             }, []);
 
-            const onClickDropDown = useCallback(() => {
-                if (!isOpen) {
-                    setQueryText('');
-                    !isOpen && setIsOpen(true);
-                }
-            }, [isOpen]);
-
             const placeholder = useMemo(() => {
                 return contextManifest?.placeholder
                     ? contextManifest.placeholder
                     : 'Search context';
             }, [contextManifest?.placeholder]);
 
+            const hasFocus = inputRef.current === document.activeElement;
+
             return (
-                <>
+                <div style={{ display: 'flex', alignItems: 'center' }}>
                     <SearchIcon color="#DADADA" />
-                    <input
-                        type="text"
-                        value={selectedValue}
-                        onChange={onChangeQueryText}
-                        onClick={onClickDropDown}
-                        onKeyUp={onKeyUpCloseDropDown}
-                        placeholder={selectedValue !== '' ? selectedValue : placeholder}
-                        className={styles.searchInput}
-                        ref={inputRef}
-                    />
+                    <div style={{ position: 'relative' }}>
+                        <input
+                            type="text"
+                            value={queryText}
+                            onFocus={() => setIsOpen(true)}
+                            onChange={onChangeQueryText}
+                            onKeyUp={onKeyUpCloseDropDown}
+                            placeholder={placeholder}
+                            style={{ opacity: hasFocus ? 1 : 0 }}
+                            className={styles.searchContainer}
+                            ref={inputRef}
+                        />
+                        <span
+                            className={classNames(styles.searchInput, styles.overlay)}
+                            style={{
+                                opacity: hasFocus ? 0 : 1,
```

Comment:
Avoid inline styling

Reply:
inline styling is not bad, this is better than adding 2 classnames for just toggling opacity.
when react parse the style from the virtual dom to the browser dom, this inline style would be shorter than the hashed generated classname for this element. Also the browser need to do a lookup for the marked class of the element and apply style, consider all the garbage style generated inside head, this is a costly operation. 

In the situation where you have multiple elements, like td inside a table, the parse size would most likely be greater and therefor have a longer render time on update.

normally one would just select the element and toggle the css property, but in React, using a Ref is also very costly.

Using classes is a good rule of thumb, especially considering reuse on multiple elements to optimize rendering, but there is also exception to this rule